webapp/graphite/finders/remote.py

```diff
@@ -95,6 +95,7 @@ def fetch(self, nodes_or_patterns, start_time, end_time, now=None, requestContex
             else:
                 patterns.append(v.path)
 
+        # TODO: Use the thread pool here.
```

Comment:
Ah right...

I feel like the mix of FetchInProgress + Threads + generators will need to be cleaned to use a single way to doing async stuff.